src/main/java/org/mule/service/http/impl/service/HttpMessageLogger.java

```diff
@@ -19,18 +19,18 @@
  */
 public class HttpMessageLogger extends HttpProbe.Adapter {
 
-  private static final Logger logger = LoggerFactory.getLogger(HttpMessageLogger.class);
-
+  private final Logger logger;
   private final LoggerType loggerType;
   private final ClassLoader classLoader;
 
   public enum LoggerType {
     LISTENER, REQUESTER
   }
 
-  public HttpMessageLogger(final LoggerType loggerType, ClassLoader classLoader) {
+  public HttpMessageLogger(final LoggerType loggerType, String identifier, ClassLoader classLoader) {
     this.loggerType = loggerType;
     this.classLoader = classLoader;
+    this.logger = LoggerFactory.getLogger(HttpMessageLogger.class.getName() + "." + identifier);
```

Comment:
My main concern here is how you'll revert back to the old behavior and get the data from all HTTP traffic, specifically because you may not necessarily know which name to use (if it's an internal usage as in oauth). Because of that change this will most likely be something for 4.2.0 (`master` branch).

Reply:
Because of how Log4j2 works, if you configure a package, all of the packages within will automatically have the same configuration. The way that it knows that is a subpackage is by separating it with dots. Basically, what I am doing is to add a subcategory so the same configuration still works, but you have more fine-grained categories in it to filter.

The following configuration will work for all the HTTP traffic, like today:
```<AsyncLogger name="org.mule.service.http.impl.service.HttpMessageLogger" level="DEBUG" />```

The following will only apply to the configuration traffic using the configuration Http_Listener_Config:
```<AsyncLogger name="org.mule.service.http.impl.service.HttpMessageLogger.Http_Listener_Config" level="DEBUG" />```

Reply:
Should I create a different PR for the master?

Reply:
Hadn’t thought of that, awesome. Regarding this PR, change it to point to master, I’ll do the cherry-picks once it’s merged and we have discussed  with the team whether to put it in 4.1.4.